src/python/pants/backend/scala/compile/scalac_plugins.py

```diff
@@ -107,6 +165,94 @@ async def global_scalac_plugins(
     return GlobalScalacPlugins(loaded_global_plugins.names, classpath)
 
 
+@rule
+async def all_scala_plugin_targets(targets: AllTargets) -> AllScalaPluginTargets:
+    return AllScalaPluginTargets(
+        tgt for tgt in targets if tgt.has_fields((ScalacPluginArtifactField, ScalacPluginNameField))
+    )
+
+
+@rule
+async def resolve_scala_plugins_for_target(
+    request: ScalaPluginsForTargetRequest,
+    all_scala_plugins: AllScalaPluginTargets,
+    jvm: JvmSubsystem,
+    scalac: Scalac,
+) -> ScalaPluginTargetsForTarget:
+
+    target = request.target
+    resolve = target[JvmResolveField].normalized_value(jvm)
+
+    plugin_names = target.get(ScalaConsumedPluginNamesField).value
+    if plugin_names is None:
```

Comment:
Could the field also just be the empty list? I assume that should be treated as a way to opt out of the default?

Reply:
Right. If explicitly offering the empty lost, then you want no plugins. None means nothing was specified in the build file.

Reply:
That makes sense, but is subtl. Worth a brief comment, plus `help` message for the field

Reply:
@Eric-Arellano OK if we leave this for a fast-follow?

Reply:
Fine with me